sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-jedis/src/main/java/com/azure/messaging/eventhubs/checkpointstore/jedis/JedisRedisCheckpointStore.java

```diff
@@ -0,0 +1,162 @@
+import com.azure.core.util.serializer.JsonSerializerProviders;
+import com.azure.core.util.serializer.ObjectSerializer;
+import com.azure.core.util.serializer.TypeReference;
+import com.azure.messaging.eventhubs.CheckpointStore;
+import com.azure.messaging.eventhubs.models.Checkpoint;
+import com.azure.messaging.eventhubs.models.PartitionOwnership;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.Jedis;
+
+/**
+ * Implementation of {@link CheckpointStore} that uses Azure Redis Cache, specifically Jedis.
+ */
+public class JedisRedisCheckpointStore implements CheckpointStore {
+    private static final ClientLogger LOGGER = new ClientLogger(JedisRedisCheckpointStore.class);
+    private final JedisPool jedisPool;
+//    private final JacksonAdapter jacksonAdapter = new JacksonAdapter();
+    private static final JsonSerializer DEFAULT_SERIALIZER = JsonSerializerProviders.createInstance(true);
+    JedisRedisCheckpointStore(JedisPool jedisPool) {
+        this.jedisPool = jedisPool;
+
+    }
+
+    /**
+     * This method returns the list of partitions that were owned successfully.
+     *
+     * @param requestedPartitionOwnerships List of partition ownerships from the current instance
+     * @return Null
+     */
+    @Override
+    public Flux<PartitionOwnership> claimOwnership(List<PartitionOwnership> requestedPartitionOwnerships) {
+        return null;
+    }
+
+    /**
+     * This method returns the list of checkpoints from the underlying data store, and if no checkpoints are available, then it returns empty results.
+     *
+     * @param fullyQualifiedNamespace The fully qualified namespace of the current instance of Event Hub
+     * @param eventHubName The Event Hub name from which checkpoint information is acquired
+     * @param consumerGroup The consumer group name associated with the checkpoint
+     * @return Null
+     */
+    @Override
+    public Flux<Checkpoint> listCheckpoints(String fullyQualifiedNamespace, String eventHubName, String consumerGroup) {
+        String prefix = prefixBuilder(fullyQualifiedNamespace, eventHubName, consumerGroup);
+        ObjectSerializer serializer = DEFAULT_SERIALIZER;
+
+        try (Jedis jedis = jedisPool.getResource()) {
+            Set<String> members = jedis.smembers(prefix);
+            if (members.isEmpty()) {
+                return Flux.error(new IllegalArgumentException()); // no checkpoints to list
+            }
+            ArrayList<Checkpoint> list = new ArrayList<>();
+            for (String member : members) {
+                //get the associated JSON representation for each for the members
+                List<String> checkpointJsonList = jedis.hmget(member, "checkpoint");
```

Comment:
These special field names should be stored as a constant. So, if we update this field name, we can refactor -> rename with the help of intellisense rather than manually looking for instances.

Reply:
You can make it package-private so that your tests can leverage this constant.